open_instruct/utils.py

```diff
@@ -56,19 +56,25 @@ def is_openai_format(messages: Any) -> bool:
 
 
 # functions for handling different formats of messages
-def instruction_output_to_messages(example):
+def convert_alpaca_gpt4_to_messages(example):
```

Comment:
Why not leaving these as general, as we can use them with multiple datasets?

Reply:
This is to mirror how they're currently being used -- none of the functions I renamed are shared right now. On the other hand I left `conversations_to_messages` as is. I also had to edit a few of the functions to add dataset-specific prompts. These will continue to evolve as we make changes to the mix, so the design will be in flux for a bit.

Reply:
I know is just sad to go from general design to specific, just for naming, and expect to need to change back to general 🙃

Reply:
Candidly, I'm not expecting to change any of these back to general 🙂 Each dataset tends to have its own logic if it's not already in our format, so I expect we'll have a handful of dataset-specific functions and that's it.

Dataset-specific is also cleaner in general, because adding a new dataset requires no changes to preexisting logic. You either re-use a function, or just write a short new one.

Reply:
Many datasets have the same logic....

Reply:
If we want to avoid this logic, (not in this PR) we can maintain our own dataset. We have like maybe 4 `create_dataset.py` for different styled datasets. Good for reproduction purposes, and we can distribute these logic in many files.

Reply:
out of all the datasets we're looking at that aren't already in a messages format, only 3 have the same logic, and they're all in the `conversations` format. all of the others are unique. like I said earlier, if a new dataset shares the same logic as a preexisting dataset, they can share a function. otherwise, each having their own is less brittle.

Reply:
> If we want to avoid this logic, (not in this PR) we can maintain our own dataset. We have like maybe 4 `create_dataset.py` for different styled datasets. Good for reproduction purposes, and we can distribute these logic in many files.
> 
> E.g., https://github.com/huggingface/trl/tree/main/examples/datasets
> 
> Easy for people to add new datasets, too.

Agreed, this is also a good solution (for a separate PR).